sources/csv_reader/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""
+Source, which uses the `filesystem` source and DuckDB
+to extract CSV files data from the given locations.
+"""
+from typing import Iterable, List, Union
+
+import duckdb
+import fsspec
+import pendulum
+
+import dlt
+from dlt.common.typing import TDataItem
+from dlt.common.storages.fsspec_filesystem import prepare_fsspec_args
+from dlt.common.storages.configuration import FilesystemConfiguration
+
+try:
+    from filesystem import filesystem
+    from filesystem import FileItemDict
+    from filesystem.helpers import (
+        AbstractFileSystem,
+        FilesystemConfigurationResource,
+        FileSystemCredentials,
+    )
+except ImportError:
+    from sources.filesystem import FileItemDict, filesystem
+    from sources.filesystem.helpers import (
+        AbstractFileSystem,
+        FilesystemConfigurationResource,
+        FileSystemCredentials,
+    )
+
+
+@dlt.resource(spec=FilesystemConfigurationResource)
+def read_location(
+    files: List[FileItemDict],
```

Comment:
please allow for the following configuration (after converting to transformer, see other comments:)
- let people pick if they want to use arrow or json. item format. some people may not want arrow because it is a huge dependency (I know you had a json version, so we need to restore it... also allow to read in batches, there's support for that in duckdb)
- allow to pass chunk size
- allow to pass kwargs that will be forwarded to duckdb read_csv so people have access to all the options

Reply:
Okay, done